validator/sawtooth_validator/journal/chain.py

```diff
@@ -777,7 +777,7 @@ def on_block_validated(self, commit_new_block, result):
                                                        result["cur_chain"])
 
                         # make sure old chain is in the block_caches
-                        for block in result["cur_chain"]:
+                        for block in reversed(result["cur_chain"]):
```

Comment:
Seems like this chain should be handed to the block cache in total, and let the block cache worry about how to order it.